File: drills/ts/harnesses/nc96_stock_with_cooldown.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.maxProfit !== "function") throw new Error("__signature_mismatch__");
  return [
    ["maxProfit([1, 2, 3, 0, 2])", show(3), show(solution.maxProfit([1, 2, 3, 0, 2]))],
    ["maxProfit([1])", show(0), show(solution.maxProfit([1]))],
    ["maxProfit([])", show(0), show(solution.maxProfit([]))],
    ["maxProfit([2, 1])", show(0), show(solution.maxProfit([2, 1]))],
    ["maxProfit([1, 2, 3, 4, 5])", show(4), show(solution.maxProfit([1, 2, 3, 4, 5]))],
    ["maxProfit([6, 1, 3, 2, 4, 7])", show(6), show(solution.maxProfit([6, 1, 3, 2, 4, 7]))],
  ];
}

File: drills/go/harnesses/nc96_stock_with_cooldown.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("maxProfit([1,2,3,0,2])", 3, maxProfit([]int{1, 2, 3, 0, 2})),
			tc("maxProfit([1])", 0, maxProfit([]int{1})),
			tc("maxProfit([2,1])", 0, maxProfit([]int{2, 1})),
			tc("maxProfit([1,2,4])", 3, maxProfit([]int{1, 2, 4})),
			tc("maxProfit([6,1,3,2,4,7])", 6, maxProfit([]int{6, 1, 3, 2, 4, 7})),
		}
	})
}
